Build fails on arm64: error: 'regparm' is not valid on this platform

```
In file included from /wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/algorithms/CMakeFiles/pika_algorithms.dir/Unity/unity_0_cxx.cxx:8:
In file included from /wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/algorithms/src/task_group.cpp:11:
In file included from /wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/algorithms/include/pika/parallel/task_group.hpp:19:
In file included from /wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/executors/include/pika/executors/parallel_executor.hpp:16:
In file included from /wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/execution/include/pika/execution/detail/async_launch_policy_dispatch.hpp:16:
In file included from /wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/futures/include/pika/futures/future.hpp:22:
In file included from /wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/futures/include/pika/futures/detail/future_data.hpp:21:
In file included from /wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/synchronization/include/pika/synchronization/condition_variable.hpp:16:
In file included from /wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/synchronization/include/pika/synchronization/mutex.hpp:14:
In file included from /wrkdirs/usr/ports/devel/pika/work/.build/libs/pika/threading_base/include/pika/modules/threading_base.hpp:18:
In file included from /wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/threading_base/include/pika/threading_base/execution_agent.hpp:11:
In file included from /wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/coroutines/include/pika/coroutines/detail/coroutine_impl.hpp:41:
In file included from /wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/coroutines/include/pika/coroutines/detail/context_base.hpp:44:
In file included from /wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/coroutines/include/pika/coroutines/detail/context_impl.hpp:118:
/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/coroutines/include/pika/coroutines/detail/context_linux_x86.hpp:75:18: error: 'regparm' is not valid on this platform
    __attribute((regparm(2)));
                 ^       ~
/wrkdirs/usr/ports/devel/pika/work/pika-0.6.0/libs/pika/coroutines/include/pika/coroutines/detail/context_linux_x86.hpp:77:18: error: 'regparm' is not valid on this platform
    __attribute((regparm(2)));
                 ^       ~
2 errors generated.
```

Full log: http://ampere3.nyi.freebsd.org/data/130arm64-default/a51002eeb3d0/logs/pika-0.6.0.log (IPv6 URL).

clang-13.1
FreeBSD 13.1

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -624,7 +624,6 @@ jobs:
                 -DPIKA_WITH_MALLOC=system \
                 -DPIKA_WITH_MPI=ON \
                 -DPIKA_WITH_EXAMPLES=On \
-                -DPIKA_WITH_GENERIC_CONTEXT_COROUTINES=On \
                 -DPIKA_WITH_TESTS=On \
                 -DPIKA_WITH_TESTS_HEADERS=On \
                 -DPIKA_WITH_PARALLEL_TESTS_BIND_NONE=On \

--- a/libs/pika/coroutines/include/pika/coroutines/detail/context_impl.hpp
+++ b/libs/pika/coroutines/include/pika/coroutines/detail/context_impl.hpp
@@ -113,7 +113,8 @@ namespace pika { namespace threads { namespace coroutines { namespace detail {
 }}}}    // namespace pika::threads::coroutines::detail
 
 #elif (defined(__linux) || defined(linux) || defined(__linux__)) &&            \
-    !defined(__bgq__) && !defined(__powerpc__) && !defined(__s390x__)
+    !defined(__bgq__) && !defined(__powerpc__) && !defined(__s390x__) &&       \
+    !defined(__arm__) && !defined(__arm64__) && !defined(__aarch64__)
 
 #include <pika/coroutines/detail/context_linux_x86.hpp>
 namespace pika { namespace threads { namespace coroutines { namespace detail {
@@ -122,7 +123,8 @@ namespace pika { namespace threads { namespace coroutines { namespace detail {
 }}}}    // namespace pika::threads::coroutines::detail
 
 #elif defined(_POSIX_VERSION) || defined(__bgq__) || defined(__powerpc__) ||   \
-    defined(__s390x__)
+    defined(__s390x__) || defined(__arm__) || defined(__arm64__) ||            \
+    defined(__aarch64__)
 
 #include <pika/coroutines/detail/context_posix.hpp>
 namespace pika { namespace threads { namespace coroutines { namespace detail {
@@ -140,6 +142,8 @@ namespace pika { namespace threads { namespace coroutines { namespace detail {
 
 #else
 
-#error No default_context_impl available for this system
+#error No default context switching implementation available for this system. \
+    Your platform may be supported with the CMake option \
+    PIKA_WITH_GENERIC_CONTEXT_COROUTINES=ON (requires Boost.Context).
 
 #endif    // PIKA_HAVE_GENERIC_CONTEXT_COROUTINES

--- a/libs/pika/coroutines/include/pika/coroutines/detail/context_posix.hpp
+++ b/libs/pika/coroutines/include/pika/coroutines/detail/context_posix.hpp
@@ -285,8 +285,7 @@ namespace pika::threads::coroutines {
 
 #define COROUTINE_STACKOVERFLOW_ADDR_EPSILON 1000UL
 
-            static void sigsegv_handler(
-                int signum, siginfo_t* infoptr, void* ctxptr)
+            static void sigsegv_handler(int, siginfo_t* infoptr, void* ctxptr)
             {
                 ucontext_t* uc_ctx = static_cast<ucontext_t*>(ctxptr);
                 char* sigsegv_ptr = static_cast<char*>(infoptr->si_addr);

--- a/libs/pika/coroutines/src/detail/context_posix.cpp
+++ b/libs/pika/coroutines/src/detail/context_posix.cpp
@@ -15,12 +15,14 @@
 // left empty on purpose
 
 #elif (defined(__linux) || defined(linux) || defined(__linux__)) &&            \
-    !defined(__bgq__) && !defined(__powerpc__) && !defined(__s390x__)
+    !defined(__bgq__) && !defined(__powerpc__) && !defined(__s390x__) &&       \
+    !defined(__arm__) && !defined(__arm64__) && !defined(__aarch64__)
 
 // left empty on purpose
 
 #elif defined(_POSIX_VERSION) || defined(__bgq__) || defined(__powerpc__) ||   \
-    defined(__s390x__)
+    defined(__s390x__) || defined(__arm__) || defined(__arm64__) ||            \
+    defined(__aarch64__)
 
 #include <cstddef>
 

--- a/libs/pika/coroutines/src/swapcontext.cpp
+++ b/libs/pika/coroutines/src/swapcontext.cpp
@@ -10,7 +10,8 @@
 #if !defined(PIKA_HAVE_GENERIC_CONTEXT_COROUTINES)
 
 #if (defined(__linux) || defined(linux) || defined(__linux__)) &&              \
-    !defined(__bgq__) && !defined(__powerpc__) && !defined(__s390x__)
+    !defined(__bgq__) && !defined(__powerpc__) && !defined(__s390x__) &&       \
+    !defined(__arm__) && !defined(__arm64__) && !defined(__aarch64__)
 
 #if defined(__x86_64__) || defined(__amd64__)
 #include "swapcontext64.ipp"
@@ -20,7 +21,7 @@
 #else
 #error You are trying to use x86 context switching on a non-x86 platform. Your \
     platform may be supported with the CMake option \
-    PIKA_WITH_GENERIC_CONTEXT_COROUTINES=ON.
+    PIKA_WITH_GENERIC_CONTEXT_COROUTINES=ON (requires Boost.Context).
 #endif
 
 #endif

